chain/chain/src/chain.rs

```diff
@@ -777,7 +790,9 @@ impl<'a> ChainUpdate<'a> {
             return Err(ErrorKind::InvalidBlockFutureTime(header.timestamp).into());
         }
 
-        // First I/O cost, delayed as late as possible.
+        // First I/O cost, delay as much as possible.
+        self.check_header_signature(header)?;
```

Comment:
Extract the actual logic for the signature from `compute_weight`.
Ask Illia why it is there to begin with.
Open an issue to fix the implmenentation of `check_validator_signature`